hub/authorization.go

```diff
@@ -71,10 +74,28 @@ func authorize(r *http.Request, jwtKey []byte, publishAllowedOrigins []string) (
 // validateJWT validates that the provided JWT token is a valid Mercure token
 func validateJWT(encodedToken string, key []byte) (*claims, error) {
 	token, err := jwt.ParseWithClaims(encodedToken, &claims{}, func(token *jwt.Token) (interface{}, error) {
-		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
-			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		switch token.Method.(type) {
```

Comment:
Instead of relying on the algorithm selected by the client, shouldn't we create a new environment variable to configure the algorithm (having `HMAC ` as default) and always use the configured one?

Reply:
The algorithm is not selected by the client, but the identity provider signing the jwt.
IMO we should be able to trust the selected method. It must be compatible with the private key anyways.

Reply:
The client can forge a JWT token and then select the algorithm to use.

https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/

>  In any case, I don't think JWT libraries should even look at the alg field in the header, except maybe to check that it matches what was the expected algorithm.
> Anyone using a JWT implementation should make sure that tokens with a different signature type are guaranteed to be rejected.

There is no reason to trigger a RSA check if the a HMAC key is configured (and vice versa).

Reply:
You are right, haven't thought of this.

Algo none is forbidden, but forging the public key into an HMAC signed JWT was still present.

Added new commits and rebased my branch.